packages/cli/src/commands/contract/tx.ts

```diff
@@ -0,0 +1,76 @@
+import { Flags } from "@oclif/core";
+import { ContractPromise } from "@polkadot/api-contract";
+import { cryptoWaitReady } from "@polkadot/util-crypto";
+
+import { ContractCall } from "../../lib/contractCall";
+
+export class Tx extends ContractCall<typeof Tx> {
+  static summary = "Call a Tx message on smart contract";
+
+  static flags = {
+    dry: Flags.boolean({
+      char: "d",
+      description: "Do a dry run, without signing the transaction",
+    }),
+    account: Flags.string({
+      required: true,
+      char: "a",
+      description: "Account to sign the transaction with",
+    }),
+  };
+
+  static args = [...ContractCall.callArgs];
+
+  public async run(): Promise<void> {
+    const { flags, args } = await this.parse(Tx);
+
+    const contract = new ContractPromise(
+      this.api.apiInst,
+      this.metadata,
+      this.deploymentInfo.address
+    );
+
+    const storageDepositLimit = null;
+
+    const queryResult = await contract.query[args.messageName](
+      this.account.pair.address,
+      {
+        gasLimit: this.api.apiInst.registry.createType("WeightV2", {
+          refTime: BigInt(10000000000),
+          proofSize: BigInt(10000000000),
+        }),
+        storageDepositLimit,
+      },
+      ...flags.params
+    );
+
+    this.log(`Gas required: ${queryResult.gasRequired.toHuman()}`);
```

Comment:
When getting weightV2, log shows `[object Object]`.
If this is weight V2, better to show refTime and proofSize.
<img width="1146" alt="Screen Shot 2023-01-23 at 10 31 04" src="https://user-images.githubusercontent.com/28953860/213952395-342ad9f8-1034-4514-8da7-29777db28ddf.png">